### PR TITLE
Storybook: Add missing select options for controls

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:

ConfirmButton.story and RadioButtonGroup.story were missing options for select controls, causing them to crash. Also according to the [Storybook's docs](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-controloptions), the options are not part of the `control` prop in the `argTypes` anymore.

**Which issue(s) this PR fixes**:

Contributes to https://github.com/grafana/grafana/issues/34417

### CODE REVIEW

File: packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx [hunk 1]
```diff
@@ -33,8 +33,13 @@ export default {
     closeOnConfirm: true,
   },
   argTypes: {
-    confirmVariant: { control: { type: 'select' } },
-    size: { control: { type: 'select' } },
+    confirmVariant: {
+      control: {
+        type: 'select',
+      },
+      options: ['primary', 'secondary', 'destructive', 'link'],
```
Comment (jackw) [inline packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx:40]: I was under the impression Storybook would infer these from the types.

File: packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx [hunk 2]
```diff
@@ -43,6 +48,7 @@ interface StoryProps extends Partial<Props> {
 }
 
 export const Basic: Story<StoryProps> = (args) => {
+  console.log('var', args.confirmVariant);
```
Comment (jackw) [inline packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx:51]: Is this a stray debug?

Comment (Clarity-89) [reply packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx:51]: Good catch! 🙈

File: packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/ConfirmButton/ConfirmButton.story.tsx:40]: Doesn't seem like it: https://github.com/storybookjs/storybook/issues/13380#issuecomment-739104528

**Review by thisisobate — APPROVED** [review]
LGTM!
I think the only thing left is to exclude unnecessary controls such as `onClick`, `onCancel`, `onConfirm` for the confirmButton Story, and `onChange` for the RadioButtonGroup Story.

**Comment by thisisobate** [comment]
> LGTM!
> I think the only thing left is to exclude unnecessary controls such as `onClick`, `onCancel`, `onConfirm` for the confirmButton Story, and `onChange` for the RadioButtonGroup Story.

NVM...Would fix all of those in a future PR. 👍

**Pull request merged**